### PR TITLE
Update CA doc links to point to dotnet docs repo

### LINKED ISSUE #4286: Update help uri to point to docs.microsoft.com/dotnet
The documentation for CA rules has moved to .NET docs instead of Visual Studio docs. The help uri for them should point to the new location.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #4286

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md [hunk 1]
```diff
@@ -2,5 +2,6 @@
 
 Rule ID | Missing Help Link | Title |
 --------|-------------------|-------|
-CA1416 | https://docs.microsoft.com/visualstudio/code-quality/ca1416 | Validate platform compatibility |
+CA2218 | https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2218 | Override GetHashCode on overriding Equals |
+CA2224 | https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2224 | Override Equals on overloading operator equals |
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md:6]: @gewarren seems like we missed porting docs for `CA2218` and `CA2224` to dotnet docs repo. The pages still exist in VS docs repo, though these have been ported as analyzers. Likely I missed them in the initial work.

File: src/Utilities/Compiler/DiagnosticDescriptorHelper.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ internal static partial class DiagnosticDescriptorHelper
             var (defaultSeverity, enabledByDefault) = GetDefaultSeverityAndEnabledByDefault(ruleLevel, isEnabledByDefaultInFxCopAnalyzers);
 
 #pragma warning disable CA1308 // Normalize strings to uppercase - use lower case ID in help link
-            var helpLink = $"https://docs.microsoft.com/visualstudio/code-quality/{id.ToLowerInvariant()}";
+            var helpLink = $"https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/{id.ToLowerInvariant()}";
```
Comment (mavasani) [inline src/Utilities/Compiler/DiagnosticDescriptorHelper.cs:36]: This is the only manual change in this PR. Rest all are automatic updates from repo tooling on running pack.

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md:6]: Ah, seems like I missed `CA2218` as it was converted to a VB only analyzer: https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/NetAnalyzers/VisualBasic/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/BasicOverrideGetHashCodeOnOverridingEquals.vb#L11-L16

C# does not need it.

Comment (mavasani) [reply src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md:6]: Same for `CA2224`: https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/NetAnalyzers/VisualBasic/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/BasicOverrideEqualsOnOverloadingOperatorEquals.vb#L11-L16

**Review by Evangelink — APPROVED** [review]

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.FxCopAnalyzers/RulesMissingDocumentation.md:6]: Filed https://github.com/dotnet/docs/issues/20917 for doc porting.

**Comment by mavasani** [comment]
Thanks @Evangelink

**Review by Youssef1313 — COMMENTED** [review]
@mavasani There are some files where the old links are still there (not all of them are auto-generated, but **some of them are**).

https://github.com/dotnet/roslyn-analyzers/search?l=Markdown&p=1&q=visualstudio

**Pull request merged**